src/sentry/features/manager.py

```diff
@@ -41,24 +38,37 @@ def get(self, name, *args, **kwargs):
         >>> FeatureManager.get('my:feature', actor=request.user)
         """
         try:
-            cls = self._registry[name]
+            cls = self._feature_registry[name]
         except KeyError:
             raise FeatureNotRegistered(name)
         return cls(name, *args, **kwargs)
 
+    def add_handler(self, handler):
+        """
+        Register a feature handler.
+
+        The passed object is a FeatureHandler that is not associated with any
+        one feature name, but will be applied to every flag that the ``has``
+        method checks.
+        """
+        for feature_name in handler.features:
+            handlers = self._handler_registry.get(feature_name, set())
```

Comment:
Nit: You could make these a `collections.defaultdict(set)` and then just do `self._handler_registry[feature_name].add(handler)`

Reply:
Oops, forgot that was a thing. Good catch.